app/lib/zif/services/input_service.rb

```diff
@@ -123,6 +139,33 @@ def process_click
       end
       # rubocop:enable Metrics/PerceivedComplexity
 
+      # @api private
+      def process_key_event
+        return if @key_pressables.empty?
+
+        key_down = $gtk.args.inputs.keyboard.key_down.truthy_keys
+        # sometimes it's [:char, :raw_key, :s, :shift]
+        # sometimes it's [:char, :raw_key, :shift, :d]
+        # sometimes it's [:char, :raw_key, :shift_left, :d, :control_left, :control]
+        # so let's not care about shifts
+        key_down.delete(:shift)
+        return if key_down.empty?
+
+        if key_down == %i[char raw_key]
+          # this is a repeat char, aaaaaa, leave @last_key alone
+        elsif key_down[0] == :char # [:char, :raw_key, :s]
+          @last_key = key_down[2]
+        elsif key_down[0] == :raw_key # [:raw_key, :shift_left]
+          @last_key = key_down[1]
+        end
```

Comment:
- Replace noop if statement with nested if
- Replace first and last index gets with `first` and `last` - check if this is appropriate?
```suggestion
        if key_down != %i[char raw_key]
          if key_down.first == :char # [:char, :raw_key, :s]
            @last_key = key_down.last
          elsif key_down.first == :raw_key # [:raw_key, :shift_left]
            @last_key = key_down[1]
          end
        end
```